src/NetStream.Specs/Specifications/Integration/ConsumeSpecs.cs

```diff
@@ -126,5 +128,88 @@ class when_consuming_a_message_with_auto_commit_enabled
+                })
+                .Stream(_sourceTopic)
+                .Handle(context =>
+                {
+                    _consumedMessages.Add(context.Message);
+                    throw new Exception("Bang!");
+                })
+                .ToTopic<string, TestMessage>(_destinationTopic)
+                .Build()
+                .StartAsync(CancellationToken.None);
+
+                _producer.Produce(Guid.NewGuid().ToString(), new TestMessage());
+            };
+
+            Because of = () => _producer.ProduceAsync(Guid.NewGuid().ToString(), new TestMessage()).BlockUntil(() => _consumedMessages.Count == 2).Await();
+
+            It should_continue_processing_the_next_message = () => _consumedMessages.Count.ShouldEqual(2);
+        }
+
+        [Subject("Consume Failure")]
+        class when_consuming_a_message_and_configured_to_stop_on_error
+        {
+            static readonly string _sourceTopic = $"f.{Guid.NewGuid()}";
+            static readonly string _destinationTopic = $"f.{Guid.NewGuid()}";
+            static TestProducerService<string, TestMessage> _producer;
+            static INetStream _stream;
+            static List<TestMessage> _consumedMessages = new List<TestMessage>();
+            static Task _streamTask;
+
+            Establish context = () =>
+            {
+                new TopicService().CreateAll(_sourceTopic, _destinationTopic);
+
+                _producer = new TestProducerService<string, TestMessage>(_sourceTopic);
+                _stream = new NetStreamBuilder<string, TestMessage>(cfg =>
+                        {
+                            cfg.BootstrapServers = "localhost:9092";
+                            cfg.ConsumerGroup = Guid.NewGuid().ToString();
+                            cfg.AutoOffsetReset = AutoOffsetReset.Earliest;
+                            cfg.ContinueOnError = false;
+                        })
+                        .Stream(_sourceTopic)
+                        .Handle(context =>
+                        {
+                            _consumedMessages.Add(context.Message);
+                            throw new Exception("Bang!");
+                        })
+                        .ToTopic<string, TestMessage>(_destinationTopic)
+                        .Build();
+
+                _producer.Produce(Guid.NewGuid().ToString(), new TestMessage());
+                _producer.Produce(Guid.NewGuid().ToString(), new TestMessage());
+
+                _streamTask = _stream.StartAsync(CancellationToken.None);
+            };
+
+            Because of = () => _streamTask.Wait();
+
+            It should_not_continue_processing_the_next_message = () => _consumedMessages.Count.ShouldEqual(1);
+
+            It should_stop_the_stream = () => _stream.Status.ShouldEqual(NetStreamStatus.Stopped);
+
+            It should_complete_the_stream_task = () => _streamTask.Status.ShouldEqual(TaskStatus.RanToCompletion);
+        }
     }
```

Comment:
Might be a good idea to add a test for Starting a Stream after it has stopped. I can see someone doing this. It looks like it probably will just work, but we might get something unexpected from calling Subscribe or CreateAllTopics a second time.